pkg/agent/controller/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,417 @@
+			case watch.Added:
+				c.enqueueSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.InfoS("Enqueue added SupportBundleCollection", "name", event.Object.(*cpv1b2.SupportBundleCollection).Name)
+			case watch.Deleted:
+				c.enqueueSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.InfoS("Enqueue removed SupportBundleCollection", "name", event.Object.(*cpv1b2.SupportBundleCollection).Name)
+			default:
+				klog.Errorf("Unknown event: %v", event)
+				return
+			}
+			eventCount++
+		}
+
+	}
+}
+
+func (c *SupportBundleController) enqueueSupportBundleCollection(supportbundle *cpv1b2.SupportBundleCollection) {
+	c.queue.Add(supportbundle.Name)
+}
+
+func (c *SupportBundleController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.InfoS("Starting", "controllerName", controllerName)
+	defer klog.InfoS("Shutting down", "controllerName", controllerName)
+
+	go wait.NonSlidingUntil(c.watchSupportBundleCollections, 5*time.Second, stopCh)
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (c *SupportBundleController) GetSupportBundleCollectionCondition() v1alpha1.SupportBundleCollectionConditionType {
+	c.conditionLock.RLock()
+	defer c.conditionLock.RUnlock()
+	return c.lastCondition
+}
+
+func (c *SupportBundleController) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *SupportBundleController) processNextWorkItem() bool {
+	obj, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(obj)
+
+	if key, ok := obj.(string); !ok {
+		klog.Errorf("Expected string in work queue but got %#v", obj)
+		return true
+	} else if err := c.syncSupportBundleCollection(key); err == nil {
+		klog.InfoS("Successfully synced support bundle", "name", key)
+	} else {
+		// Skip retrying as the time may not meet the requirements for SupportBundle.
+		klog.ErrorS(err, "Error syncing SupportBundleCollection", "name", key)
+	}
+	return true
+}
+
+func (c *SupportBundleController) syncSupportBundleCollection(key string) error {
+	klog.InfoS("Sync support bundle collection", "name", key)
+	antreaClient, err := c.antreaClientGetter.GetAntreaClient()
+	if err != nil {
+		return fmt.Errorf("failed to get antrea client: %w", err)
+	}
+	supportBundle, err := antreaClient.ControlplaneV1beta2().SupportBundleCollections().Get(context.TODO(), key, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Ignore support bundle deletion event.
+		klog.InfoS("Skip syncing support bundle collection because the support bundle has been deleted", "name", key)
+		return nil
+	}
+	err = c.generateSupportBundle(supportBundle)
+	c.conditionLock.Lock()
+	defer c.conditionLock.Unlock()
+	var nodeType string
+	if c.nodeType == config.K8sNode {
+		nodeType = string(controlplane.SupportBundleCollectionNodeTypeNode)
+	} else {
+		nodeType = string(controlplane.SupportBundleCollectionNodeTypeExternalNode)
+	}
```

Comment:
does this need to be locked?

Reply:
Have changed this part, please help check the latest version.